optuna/storages/_base.py

```diff
@@ -294,7 +294,7 @@ def get_study_system_attrs(self, study_id: int) -> Dict[str, Any]:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_all_study_summaries(self) -> List[StudySummary]:
+    def get_all_study_summaries(self, include_best_trial: bool = True) -> List[StudySummary]:
```

Comment:
[question]

The same default value (`True` for `include_best_trial`) are both added in `optuna.storages.BaseStorage.get_all_study_summaries` and `optuna.get_all_study_summaries`. I'm not sure if it is a good practice.

Is it impossible to remove the default value from storage `get_all_study_summaries` methods?This should be breaking change but it would be acceptable if users don't invoke this method directly but invoke `optuna.get_all_study_summaries`, I think.

Reply:
Thank you for your suggestion. That makes sense.
The `optuna.get_all_study_summaries` is public API, and users basically use it instead of  `optuna.storages.BaseStorage.get_all_study_summaries`.

`optuna.storages.RDBStorage` is listed in the reference page, but [the reference](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.storages.RedisStorage.html) does not encourage to call it.

> Note that library users can instantiate this class, but the attributes provided by this class are not supposed to be directly accessed by them.

I fixed it in https://github.com/optuna/optuna/pull/3109/commits/e891e941cf6ee4b2bc509bdb5eeb12a6c251ce0c and https://github.com/optuna/optuna/pull/3109/commits/a3cd1c7d5bbc14b28a8d0effa09965079533001a. In addition, the missing document about the `include_best_trial` is added in https://github.com/optuna/optuna/pull/3109/commits/e891e941cf6ee4b2bc509bdb5eeb12a6c251ce0c.

Reply:
Thank you for your deep consideration! It looks fair enough.